swap/src/monero.rs

```diff
@@ -99,6 +99,31 @@ impl Amount {
         self.0
     }
 
+    pub fn as_monero_decimal(&self) -> Decimal {
+        let piconero_decimal = Decimal::from(self.0);
+        let pic_offcet_decimal = Self::piconero_offset_decimal();
+        piconero_decimal / pic_offcet_decimal
+    }
+
+    fn piconero_offset_decimal() -> Decimal {
+        Decimal::from(PICONERO_OFFSET)
+    }
+
+    pub fn max_bitcoin_from_balance(&self, satoshis: u64) -> Result<bitcoin::Amount> {
+        let piconero_minus_fee = self.as_piconero().saturating_sub(MONERO_FEE.as_piconero());
+
+        if piconero_minus_fee == 0 {
+            tracing::error!("The current Monero balance is smaller than the basic locking fee");
+            bail!("The current Monero balance is smaller than the basic locking fee");
+        }
```

Comment:
Wouldn't it be more logical to just return zero? When calling this function we are essential asking "How much Bitcoin can we convert this into at this exchange rate while accounting for the fee?". If the answer to that questions happens to be zero, that's not really an error imo.

Also, will the ASB return a quote at all when the balance is zero?

Reply:
the 0 is actually not the real number if I had to use a normal subtraction I would get a negative number but because its a `u64` integer I would get an integer underflow panic, so I returned this as an elegant error to the asb provider. 

For the swap app if you get request a quote from a rendezvous server you wont get one from the the particular ASB with the lower then minimum balance and if you request it directly the connection will drop.